Updated slack references and removed dead links

rdpiazza opened the pull request.

rdpiazza committed:
Updated slack references and removed dead links

suprenant reviewed (changes requested):
Getting an error for the Hub link:
404 Not Found: Requested route ('hub.18f.gov') does not exist.

rdpiazza committed:
Removed reference to the Hub

Removed reference to the Hub and nearby content.  This section is largely out of date, so not many concerns about losing content.  Content will need to be updated with more rigor in the future to reflect current processes.

rdpiazza commented:
Pushed requested change to the branch.  Reference to Hub was removed.
